sylvia/src/multitest.rs

```diff
@@ -1,13 +1,51 @@
 use std::cell::{Ref, RefCell, RefMut};
+use std::fmt::Debug;
 use std::marker::PhantomData;
 
-use cosmwasm_std::{Addr, BlockInfo, Coin};
-use cw_multi_test::Executor;
+use cosmwasm_std::{
+    Addr, Api, BlockInfo, Coin, CustomQuery, Empty, GovMsg, IbcMsg, IbcQuery, Storage,
+};
+use cw_multi_test::{
+    AppBuilder, BankKeeper, DistributionKeeper, Executor, FailingModule, Router, StakeKeeper,
+    WasmKeeper,
+};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
 use serde::Serialize;
 
-#[derive(Default)]
-pub struct App {
-    pub app: RefCell<cw_multi_test::App>,
+pub struct App<ExecC = Empty, QueryC = Empty> {
+    pub app: RefCell<cw_multi_test::BasicApp<ExecC, QueryC>>,
+}
+
+impl Default for App {
+    fn default() -> Self {
+        Self::new(cw_multi_test::BasicApp::default())
+    }
+}
+
+/// Creates new default `App` implementation working with customized exec and query messages.
+/// Outside of `App` implementation to make type elision better.
+pub fn custom_app<ExecC, QueryC, F>(init_fn: F) -> App<ExecC, QueryC>
```

Comment:
I know it is how it is done in multitest, but did you try just create an `App::custom(...)` function for more consistent API?